### ISSUE TITLE
[Bug]: exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup

### ISSUE THREAD

**Issue by chenchunhui97** [vllm-project/vllm#6630]
### Your current environment

ubuntu 22.04
cuda driver：
![image](https://github.com/user-attachments/assets/7eb39509-08e7-437f-ab67-6fa7b2b50f77)

use official image: 0.4.0 or 0.5.0post1
GPU: RTX 4090, or A800

run inference of Qwen1.5-32B-GPTQ-int4, or deepseek-coder-34B-Instruct-AWQ.
enable-prefix-caching, gpu-memory-utilization=0.9, max-model-len=8192, tensor-parallel-size=2.

### 🐛 Describe the bug

2024-07-22T10:38:57.216715364+08:00 INFO 07-22 10:38:57 async_llm_engine.py:154] Aborted request cmpl-b6da16012b2d46cdb9773cace1dad626.

2024-07-22T10:38:57.218681532+08:00 ERROR:    Exception in ASGI application

2024-07-22T10:38:57.218694849+08:00 Traceback (most recent call last):

2024-07-22T10:38:57.218703600+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/responses.py", line 265, in __call__

2024-07-22T10:38:57.218711308+08:00     await wrap(partial(self.listen_for_disconnect, receive))

2024-07-22T10:38:57.218718705+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/responses.py", line 261, in wrap

2024-07-22T10:38:57.218725837+08:00     await func()

2024-07-22T10:38:57.218733370+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/responses.py", line 238, in listen_for_disconnect

2024-07-22T10:38:57.218740597+08:00     message = await receive()

2024-07-22T10:38:57.218747573+08:00   File "/usr/local/lib/python3.10/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 553, in receive

2024-07-22T10:38:57.218754319+08:00     await self.message_event.wait()

2024-07-22T10:38:57.218761150+08:00   File "/usr/lib/python3.10/asyncio/locks.py", line 214, in wait

2024-07-22T10:38:57.218768783+08:00     await fut

2024-07-22T10:38:57.218775879+08:00 asyncio.exceptions.CancelledError: Cancelled by cancel scope 7efeac0db9a0

2024-07-22T10:38:57.218782174+08:00 

2024-07-22T10:38:57.218789239+08:00 During handling of the above exception, another exception occurred:

2024-07-22T10:38:57.218795435+08:00 

2024-07-22T10:38:57.218802106+08:00 Traceback (most recent call last):

2024-07-22T10:38:57.218808992+08:00   File "/usr/local/lib/python3.10/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 399, in run_asgi

2024-07-22T10:38:57.218815863+08:00     result = await app(  # type: ignore[func-returns-value]

2024-07-22T10:38:57.218822698+08:00   File "/usr/local/lib/python3.10/dist-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__

2024-07-22T10:38:57.218840911+08:00     return await self.app(scope, receive, send)

2024-07-22T10:38:57.218848037+08:00   File "/usr/local/lib/python3.10/dist-packages/fastapi/applications.py", line 1054, in __call__

2024-07-22T10:38:57.218854723+08:00     await super().__call__(scope, receive, send)

2024-07-22T10:38:57.218861269+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/applications.py", line 123, in __call__

2024-07-22T10:38:57.218867940+08:00     await self.middleware_stack(scope, receive, send)

2024-07-22T10:38:57.218874554+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 186, in __call__

2024-07-22T10:38:57.218881551+08:00     raise exc

2024-07-22T10:38:57.218888302+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 164, in __call__

2024-07-22T10:38:57.218894918+08:00     await self.app(scope, receive, _send)

2024-07-22T10:38:57.218901599+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/cors.py", line 85, in __call__

2024-07-22T10:38:57.218908189+08:00     await self.app(scope, receive, send)

2024-07-22T10:38:57.218914769+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/exceptions.py", line 65, in __call__

2024-07-22T10:38:57.218921500+08:00     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)

2024-07-22T10:38:57.218928542+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 64, in wrapped_app

2024-07-22T10:38:57.218935156+08:00     raise exc

2024-07-22T10:38:57.218941712+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 53, in wrapped_app

2024-07-22T10:38:57.218948453+08:00     await app(scope, receive, sender)

2024-07-22T10:38:57.218955078+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 756, in __call__

2024-07-22T10:38:57.218961714+08:00     await self.middleware_stack(scope, receive, send)

2024-07-22T10:38:57.218968334+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 776, in app

2024-07-22T10:38:57.218974999+08:00     await route.handle(scope, receive, send)

2024-07-22T10:38:57.218981711+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 297, in handle

2024-07-22T10:38:57.218988392+08:00     await self.app(scope, receive, send)

2024-07-22T10:38:57.218994967+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 77, in app

2024-07-22T10:38:57.219001528+08:00     await wrap_app_handling_exceptions(app, request)(scope, receive, send)

2024-07-22T10:38:57.219008413+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 64, in wrapped_app

2024-07-22T10:38:57.219015058+08:00     raise exc

2024-07-22T10:38:57.219021615+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 53, in wrapped_app

2024-07-22T10:38:57.219028244+08:00     await app(scope, receive, sender)

2024-07-22T10:38:57.219034946+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 75, in app

2024-07-22T10:38:57.219041650+08:00     await response(scope, receive, send)

2024-07-22T10:38:57.219048202+08:00   File "/usr/local/lib/python3.10/dist-packages/starlette/responses.py", line 258, in __call__

2024-07-22T10:38:57.219054802+08:00     async with anyio.create_task_group() as task_group:

2024-07-22T10:38:57.219061363+08:00   File "/usr/local/lib/python3.10/dist-packages/anyio/_backends/_asyncio.py", line 680, in __aexit__

2024-07-22T10:38:57.219068098+08:00     raise BaseExceptionGroup(

2024-07-22T10:38:57.219084476+08:00 exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)

**Comment by foreverhell** [comment]
I have also encountered the same issue but not solved. Waiting for solution.

**Comment by chenchunhui97** [comment]
> I have also encountered the same issue but not solved. Waiting for solution.

it seems beacuse `gpu-memory-utilization` is set too high and cause OOM. when I set `gpu-memory-utilizaton=0.9`, it reports the error, but when set `gpu-memory-utilization=0.8`, it works well.

the param `gpu-memory-utilization`   is not a hard constraint, I found, for a RTX4090 with available GPU memory=24564 M, if you set `gpu-memory-utilizaton=0.8` then the occupied memory is 21256 Mb (in my case), obviously exceeded the 0.8 constraint and when runing inference, the memory occupation further grow (it is the same when set to 0.9 and will OOM easily).
